infra/terraform/global/assets/prune_ebs_snapshots/prune_ebs_snapshots.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"./snapshot"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"log"
+	"os"
+	"strconv"
+	"time"
+)
+
+var (
+	clientSession       = session.Must(session.NewSession())
+	connection          = ec2.New(clientSession)
+	snapShotKeyFilter   = os.Getenv("SNAPSHOT_TAG_KEY")
+	snapShotValueFilter = os.Getenv("SNAPSHOT_TAG_VALUE")
+	getDays             = os.Getenv("DAYS_OLD")
+)
+
+func main() {
+
+	// Convert `DAYS_OLD` value to an integer
+	daysOldValue, err := strconv.Atoi(getDays)
+	if err != nil {
+		fmt.Println(err)
+		log.Panicf("The type of the value you provided is %T. %v needs to be an integer", daysOldValue, daysOldValue)
+	}
+
+	snapShots, err := snapshot.GetSnapShots(connection, snapShotKeyFilter, snapShotValueFilter)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	for _, snapShot := range snapShots.Snapshots {
+
+		snapShotID := *snapShot.SnapshotId
```

Comment:
Just out of curiosity, do you need to use the `*` here? Go sometimes is able to [spare you the pain](https://play.golang.org/p/lHuZHX_N0EJ) of doing it directly.

I think you already do it later when you just call `snapShot.StartTime` (without the `*`).

Where I'm getting at is that I'd probably not have this `snapShotID` variable at all and use `snapShot.SnapshotId` directly as it doesn't impact readably in fact, it's probably easier to read by removing a couple of lines of code. IMHO of course.

Reply:
The function arg is type string whereas `snapShot.SnapshotId` is `*string` so what you mean is handle it like I did the start time and change the function sig?

Reply:
Like this?
```
func PruneSnapShots(connection *ec2.EC2, snapshotID *string) (output *ec2.DeleteSnapshotOutput, err error) {

	deleteSnapShotShotInput := &ec2.DeleteSnapshotInput{
		SnapshotId: aws.String(*snapshotID),
		DryRun:     aws.Bool(true),
	}

	return connection.DeleteSnapshot(deleteSnapShotShotInput)
}
```

Reply:
I see. Nah, I think it's right that `PruneSnapshots()` takes a "normal" string.
In that case leave the client code as it is, it's probably cleaner to have the variable and it's not a big deal anyway so don't worry.